Apps/phcanary/canary_connector.py

```diff
@@ -0,0 +1,467 @@
+        # The status and progress messages are more important.
+        config = self.get_config()
+        apiKey = config.get('apiKey')
+        params = {'auth_token': apiKey}
+
+        self.save_progress("Connecting to endpoint")
+        # make rest call
+
+        ret_val, response = self._make_rest_call('/ping', action_result, params=params, headers=None, method="get")
+
+        if (phantom.is_fail(ret_val)):
+            # the call to the 3rd party device or service failed, action result should contain all the error details
+            # for now the return is commented out, but after implementation, return from here
+            self.save_progress("Test Connectivity Failed.")
+            # return action_result.get_status()
+
+        self.save_progress("Response: " + str(response))
+
+        if response:
+            if response.get('result') == "success":
+                return action_result.set_status(phantom.APP_SUCCESS)
+            else:
+                return action_result.set_status(phantom.APP_ERROR, "Error while communicating with Canary API")
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "Error while communicating with Canary API")
+
+        # For now return Error with a message, in case of success we don't set the message, but use the summary
+        # return action_result.set_status(phantom.APP_ERROR, "Action not yet implemented")
+
+    def _handle_list_incidents(self, param):
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # NOTE: test connectivity does _NOT_ take any parameters
+        # i.e. the param dictionary passed to this handler will be empty.
+        # Also typically it does not add any data into an action_result either.
+        # The status and progress messages are more important.
+        config = self.get_config()
+        apiKey = config.get('apiKey')
+        params = {'auth_token': apiKey}
+
+        incidentState = param['incident_state']
+        if incidentState == "acknowledged":
+            endpoint = "/incidents/acknowledged"
+        if incidentState == "unacknowledged":
+            endpoint = "/incidents/unacknowledged"
+
+        self.save_progress("Connecting to endpoint")
+        # make rest call
+
+        ret_val, response = self._make_rest_call(endpoint, action_result, params=params, headers=None, method="get")
+
+        if (phantom.is_fail(ret_val)):
+            # the call to the 3rd party device or service failed, action result should contain all the error details
+            # for now the return is commented out, but after implementation, return from here
+            self.save_progress("Test Connectivity Failed.")
+            # return action_result.get_status()
+
+        action_result.add_data(response)
+
+        if response:
```

Comment:
Code on line 206 can be moved into the `if` statement

Reply:
Fixed.